### DISCUSSION TITLE
handleSubmit custom params

### DISCUSSION THREAD

**Discussion by vicasas** [jaredpalmer/formik#2535] [General] [answered]
If I have a form which has two buttons to submit, one of them is to save the progress and the other is to finish the form.

How can I submit with my own logic for each of the actions in the Formik `onSubmit` event?

The idea is that each button passes or is assigned a meta data to identify the type of click that the user made to apply the logic that corresponds to each submit.

But how is this done in practice?

I had thought to props `handleSubmit` to stop a parameter by clicking the button but it doesn't work ):

````javascript
return (
  <Formik
    onSubmit={(values, formikProps, customProps) => {
        console.log(customProps) // should print partial or finish
    }}
  >
    {(handleSubmit) => (
    <input type='text' ...  />
    <button onClick={handleSubmit('partial')}>Save Progress</button>
    <button onClick={handleSubmit('finish')}>Finish</button>
    }
  <Formik>
)
```

**Comment by ypahalajani** [comment, marked as answer]
@vicasas 

A sample solution [here (link to codesandbox)](https://codesandbox.io/s/formik-custom-params-in-handlesubmit-ld8j7) 

I had followed [this (link to formik docs)](https://jaredpalmer.com/formik/docs/guides/form-submission) when I had a similar use case at work.

I hope this helps!

**Reply by anderssonq** [reply]
bro, cant see your codebox!

**Reply by ypahalajani** [reply]
@anderssonq Please check now. It got deleted by mistake. It should be recovered now.

**Reply by anderssonq** [reply]
> @anderssonq Please check now. It got deleted by mistake. It should be recovered now.

thank you bro, regards!

**Comment by bel0v** [comment]
I had a similar case where the form had 2 submit buttons with different effects. I ended up making a `submitVariant` form value that would be filled with pressing either button. So later I just looked up that value in `onSubmit` handler.
